Version 2 iujuuuu!!!

Chugeno opened the pull request:
# Enhanced Version of Domestika Downloader

Dear Rene,

First, I want to thank you for creating this amazing tool that has inspired me to add new features and improvements. I'd love to share these enhancements with you and the community!

(Note: I'm using a translator as I'm a Spanish speaker)

## Major Improvements:

### Enhanced URL Support
- Now accepts multiple URL formats (not just those ending in /courses)
- Multiple URLs can be processed in one go (separated by spaces)
- Automatically downloads all courses from the provided URLs

### macOS Compatibility
- Mac support only, I am working to make it work on windows.
- Improved file handling for macOS environment

### Interactive Cookie Management
- Interactive cookie input process
- Clear instructions on how to obtain cookies from browser
- Error handling with retry option:
  * If authentication fails, offers to input new cookie values
  * Program automatically restarts with new credentials
  * User-friendly feedback during the process

### Enhanced Video Processing
- Automatic subtitle download and embedding
- Improved video naming convention
- Better organization of downloaded content

### User Experience
- More intuitive interface
- Better error messages and handling
- Progress indicators for downloads
- Automatic retry on connection failures

## Modified Files:
- auth.js (Enhanced authentication system)
- index.js (Core functionality improvements)
- .gitignore (Better file exclusion rules)
- package.json (Updated dependencies)
- .env.example (Added for easier setup)
- README.md (Updated documentation)

## Technical Improvements:
- Better environment variable handling
- Enhanced error handling and recovery
- Improved authentication flow
- More robust download process

## Testing:
All features have been thoroughly tested on macOS and are working as expected.

I would love for you to try this version and provide any feedback. Your project has been a great inspiration, and I hope these additions can contribute to making it even better!

Thank you again for your original work that made this possible!

Best regards,
Eugenio

Chugeno committed:
Actualización mayor: modificaciones en archivos centrales y configuración

Chugeno committed:
Agregado N_m3u8DL-RE al .gitignore

Chugeno committed:
Eliminada carpeta N_m3u8DL-RE del repositorio

Chugeno committed:
Mejoras significativas: soporte macOS, múltiples URLs, manejo de cookies interactivo y subtítulos

Chugeno committed:
Eliminada carpeta N_m3u8DL-RE del repositorio (debe descargarse de la web original)

Chugeno committed:
Added bilingual README with donation links

Chugeno committed:
Updated English README with complete translation and better formatting

Chugeno committed:
Added important warning about course ownership requirement

Hermit6202 commented:
I tried it on wondows 10 but it don't work. Don't ask for URL, and donn't accept the session cookies. Only errors, errors, errors

Chugeno commented:
Hi Rene, thanks for replying and taking the time to test. Yes indeed I couldn't test what I did in windows, I have to adapt the code and the truth is that I didn't have time and I don't have a PC with windows either. I will try in the next few days to test on windows since I did it exclusively for Mac for now. I will update my README to clarify this.
As soon as I can test it I will let you know so you can try it. Thanks again for your work, it is helping me a lot.

Chugeno committed:
fix: optimized code flow and updated README files with macOS notice

Chugeno committed:
feat: major updates
- Improved code structure with mac/win separation
- Added system detection in start.js
- Removed unused dependencies
- Updated README files with macOS notice
- Cleaned up code and improved error handling
- Added proper module exports

ApratimPanwar commented:
Would love to test it on windows if possible , 
please atleast remove the hard limit that doesnt allows to do so 
"Sistema operativo detectado: Windows
Windows version is under development"

Chugeno commented:
> Would love to test it on windows if possible , please atleast remove the hard limit that doesnt allows to do so "Sistema operativo detectado: Windows Windows version is under development"

yep sorry, I haven't had time, I hope in a few months I can sit down and get this to work on windows.

Chugeno commented:
> Hola, muchas gracias @Chugeno por este gran update, funciona perfecto en MAC, solo no se bajo "proyecto final" No se si es por que no lo detecta como una unidad. Una petición ¿Es posible descargar el texto que acompaña a los videos en cada lección y/o las tareas? Muchas gracias 

Hola ralts que bueno que te funciono! He hecho pruebas de como descargar los textos/tareas y guardarlos en un word o algo asi y la verdad es muy inconsistente todavia, a veces lo hacia bien, a veces no, todavia no encontre la manera de hacerlo estable ordenado, etc... 
Y el proyecto final lo puedes ver desde la web de domestika? Por ejemplo si tienes una suscripción PLUS lo proyectos finales no estan incluidos para verlos en lo cursos disponibles con PLUS salvo que lo compres a parte se entiende? por ende no puedes descargar el proyecto final. Dime si es tu caso sino reviso que puede estar pasando.

ralts00 commented:
Hola Chugeno, gracias por contestar.
Si puedo ver desde la web de domestika, de hecho tu versión lo detecta como elegible pero cuando lo selecciono no descarga nada.
Agradezco la respuesta, un gran saludo.
<img width="1291" alt="Captura de pantalla 2025-04-16 a la(s) 20 33 16" src="https://github.com/user-attachments/assets/a77e412c-e2be-46c9-b81a-a11037f32252" />
<img width="916" alt="Captura de pantalla 2025-04-16 a la(s) 20 33 23" src="https://github.com/user-attachments/assets/08787dfd-b6ee-44bc-a8ec-d9edd70ab8ba" />
<img width="504" alt="Captura de pantalla 2025-04-16 a la(s) 20 34 30" src="https://github.com/user-attachments/assets/c2528cee-4e5c-4e75-beaf-6e58e686b8f0" />

ReneR97 commented:
Hey.
first off, thanks for all the work you put into this and sorry for the late response to this. 
But i think this would be better off if you just made a fork and released it as your own project. i dont think it fits in with what i had in mind with this script but it would be an awesome tool on its own.
